### PR TITLE
chore: create and delete nft gallery

### PR DESCRIPTION
**Pull request by sasicodes**
## What does this PR do?

Link T-3750

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Enhancement (non-breaking small changes to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- [ ] Test A
- [ ] Test B

## Checklist

- I haven't read the [contributing guide](https://github.com/lensterxyz/lenster/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't performed a self-review of my own code and corrected any misspellings
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my PR needs changes to the documentation
- I haven't checked if my changes generate no new warnings
- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by sasicodes** [comment]
Link T-3750

File: apps/web/src/components/Profile/NftGallery/Create.tsx [hunk 1]
```diff
@@ -16,40 +22,123 @@ interface Props {
 
 const Create: FC<Props> = ({ showModal, setShowModal }) => {
   const [emoji, setEmoji] = useState('');
-  const [pickNfts, setPickNfts] = useState(false);
+  const [currentStep, setCurrentStep] = useState('NAME');
```
Comment (bigint) [inline apps/web/src/components/Profile/NftGallery/Create.tsx:25]: better to have type here `<'NAME' | 'BLAH' | etc>`

File: apps/web/src/components/Profile/NftGallery/Create.tsx [hunk 2]
```diff
@@ -16,40 +22,123 @@ interface Props {
...
+          request: {
+            items: sanitizedItems,
+            name: gallery.name,
+            profileId: currentProfile?.id
+          }
+        }
+      });
+      if (result) {
+        closeModal();
+        toast.success('Gallery created');
```
Comment (bigint) [inline apps/web/src/components/Profile/NftGallery/Create.tsx:54]: Wrap in `t` macro for i18n

File: apps/web/src/components/Profile/NftGallery/Create.tsx [hunk 3]
```diff
@@ -16,40 +22,123 @@ interface Props {
...
+      }
+    } catch {}
+  };
+
+  const validateInputs = () => {
+    const galleryName = trimify(gallery.name);
+    if (!galleryName.length) {
+      toast.error('Gallery name required');
+    } else if (!gallery.items.length && currentStep === 'PICK_NFTS') {
+      toast.error('Select collectibles for your gallery');
```
Comment (bigint) [inline apps/web/src/components/Profile/NftGallery/Create.tsx:64]: Wrap in `t` macro for i18n

File: apps/web/src/components/Profile/NftGallery/Create.tsx [hunk 4]
```diff
@@ -16,40 +22,123 @@ interface Props {
...
+        closeModal();
+        toast.success('Gallery created');
+      }
+    } catch {}
+  };
+
+  const validateInputs = () => {
+    const galleryName = trimify(gallery.name);
+    if (!galleryName.length) {
+      toast.error('Gallery name required');
```
Comment (bigint) [inline apps/web/src/components/Profile/NftGallery/Create.tsx:62]: Wrap in `t` macro for i18n

File: apps/web/src/components/Profile/NftGallery/Create.tsx [hunk 5]
```diff
@@ -16,40 +22,123 @@ interface Props {
...
+    }
+  };
+
+  const onPickEmoji = (emoji: string) => {
+    setEmoji(emoji);
+    setGallery({ ...gallery, name: gallery.name + emoji });
+  };
+
+  const getModalTitle = () => {
+    if (currentStep === 'PICK_NFTS' || currentStep === 'REVIEW') {
```
Comment (bigint) [inline apps/web/src/components/Profile/NftGallery/Create.tsx:88]: Add those RHS values in enum, avoid string matching inside conditions and mainly inside JSX

File: apps/web/src/components/Profile/NftGallery/Create.tsx [hunk 6]
```diff
@@ -16,40 +22,123 @@ interface Props {
...
+              setCurrentStep(
+                currentStep === 'REVIEW' ? 'PICK_NFTS' : currentStep === 'PICK_NFTS' ? 'NAME' : 'NAME'
+              )
+            }
+          >
+            <ChevronLeftIcon className="h-4 w-4" />
+          </button>
+          <span>
+            <Trans>
+              {currentStep === 'REVIEW' ? 'Review collection' : 'Select collectibles you want others to see'}
```
Comment (bigint) [inline apps/web/src/components/Profile/NftGallery/Create.tsx:103]: Wrap in `t` macro for i18n

File: apps/web/src/components/Profile/NftGallery/Create.tsx [hunk 7]
```diff
@@ -16,40 +22,123 @@ interface Props {
...
+            </Trans>
+          </span>
+        </div>
+      );
+    }
+    return "What's your gallery name?";
```
Comment (bigint) [inline apps/web/src/components/Profile/NftGallery/Create.tsx:109]:
```suggestion
    }

    return "What's your gallery name?";
```

also wrap in `t` macro for i18n

File: apps/web/src/components/Profile/NftGallery/Create.tsx [hunk 8]
```diff
@@ -16,40 +22,123 @@ interface Props {
...
-          </div>
-        ) : (
-          "What's your gallery name?"
-        )
-      }
+      size={currentStep === 'NAME' ? 'sm' : 'lg'}
+      title={getModalTitle()}
       show={showModal}
-      onClose={() => setShowModal(false)}
+      onClose={() => closeModal()}
```
Comment (bigint) [inline apps/web/src/components/Profile/NftGallery/Create.tsx:117]: will `onClose={closeModal}` works here?

File: apps/web/src/components/Profile/NftGallery/Create.tsx [hunk 9]
```diff
@@ -16,40 +22,123 @@ interface Props {
...
+          />
+        )}
+        <div className="absolute bottom-0 flex w-full items-center justify-between rounded-b-lg bg-white p-4 dark:bg-gray-800">
+          {currentStep === 'NAME' ? (
+            <EmojiPicker emoji={emoji} setEmoji={(emoji) => onPickEmoji(emoji)} />
+          ) : (
+            `${gallery.items.length} selected`
+          )}
+          <Button disabled={loading} onClick={() => onClickNext()} icon={loading && <Spinner size="xs" />}>
+            Next
```
Comment (bigint) [inline apps/web/src/components/Profile/NftGallery/Create.tsx:139]: Wrap in `<Trans>` macro for i18n

File: apps/web/src/components/Profile/NftGallery/Gallery.tsx [hunk 10]
```diff
@@ -1,25 +1,136 @@
...
+const Gallery: FC<Props> = ({ galleries }) => {
+  const [isRearrange, setIsRearrange] = useState(false);
+  const currentProfile = useAppStore((state) => state.currentProfile);
+
+  const gallery = galleries[0];
+  const nfts = gallery.items;
+
+  const [deleteNftGallery] = useDeleteNftGalleryMutation({
+    onCompleted: () => {
+      toast.success('Gallery deleted');
```
Comment (bigint) [inline apps/web/src/components/Profile/NftGallery/Gallery.tsx:28]: Wrap in `t` macro for i18n

File: apps/web/src/components/Profile/NftGallery/Gallery.tsx [hunk 11]
```diff
@@ -1,25 +1,136 @@
...
+      location.reload();
```
Comment (bigint) [inline apps/web/src/components/Profile/NftGallery/Gallery.tsx:29]: Can we avoid this and instead remove from apollo cache

File: apps/web/src/components/Profile/NftGallery/Gallery.tsx [hunk 12]
```diff
@@ -1,25 +1,136 @@
...
+    }
+  });
+
+  const onDelete = async () => {
+    try {
+      if (confirm('Are you sure you want to delete?')) {
```
Comment (bigint) [inline apps/web/src/components/Profile/NftGallery/Gallery.tsx:35]: Wrap in `t` macro for i18n

File: apps/web/src/components/Profile/NftGallery/Gallery.tsx [hunk 13]
```diff
@@ -1,25 +1,136 @@
...
-          className="break-inside flex w-full items-center overflow-hidden text-white"
-        >
-          <NftCard nft={nft as Nft} linkToDetail={true} />
+    <>
+      <div className="flex items-center justify-between">
+        <h6 className="text-lg font-medium">{isRearrange ? 'Arrange gallery' : gallery.name}</h6>
+        {isRearrange ? (
+          <div className="flex items-center space-x-2">
+            <Button onClick={() => setIsRearrange(false)} size="sm" variant="secondary">
+              Cancel
```
Comment (bigint) [inline apps/web/src/components/Profile/NftGallery/Gallery.tsx:55]: Wrap in `<Trans>` macro for i18n

File: apps/web/src/components/Profile/NftGallery/Gallery.tsx [hunk 14]
```diff
@@ -1,25 +1,136 @@
...
+            </Button>
+            <Button onClick={() => setIsRearrange(false)} size="sm">
+              Save
```
Comment (bigint) [inline apps/web/src/components/Profile/NftGallery/Gallery.tsx:58]: Wrap in `<Trans>` macro for i18n

**Review by bigint — CHANGES_REQUESTED** [review]
@sasicodes make sure to run `yarn i18n:extract` before merging this 🙇🏼 

and left some trivial comments

File: packages/lens/documents/queries/NftGalleries.graphql [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+      tokenId
+      chainId
+      originalContent {
+        uri
+        animatedUrl
+      }
+    }
+    createdAt
+    updatedAt
+  }
```
Comment (bigint) [inline packages/lens/documents/queries/NftGalleries.graphql:20]: are we using all fields?

File: packages/lens/codegen.yml [hunk 16]
```diff
@@ -1,4 +1,4 @@
-schema: https://api-mumbai.lens.dev
+schema: https://staging-api-social-mumbai.lens.crtlkey.com
```
Comment (bigint) [inline packages/lens/codegen.yml:1]: we need to revert this before merging 🙇🏼

File: apps/web/src/components/Profile/NftGallery/ReviewSelection.tsx [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+      </div>
+    );
+  }
+
+  return (
+    <div className="grid grid-cols-1 gap-4 p-5 sm:grid-cols-3">
+      {gallery.items?.map((item) => {
+        const id = `${item.chainId}_${item.contractAddress}_${item.tokenId}`;
+        return (
+          <div key={id}>
```
Comment (bigint) [inline apps/web/src/components/Profile/NftGallery/ReviewSelection.tsx:38]: we can directly use the id and avoid return statement

**Review by bigint — CHANGES_REQUESTED** [review]

File: packages/lens/documents/queries/NftGalleries.graphql (hunk 15 above)
Comment (sasicodes) [reply packages/lens/documents/queries/NftGalleries.graphql:20]: yes except timestamps use them, same as NftFeed.

removing timestamps

**Review by bigint — APPROVED** [review]
**LGTM** :shipit: 

Rad LGTM

**Pull request merged**